cmd/kubeadm/app/cmd/init.go

```diff
@@ -309,6 +309,19 @@ func (i *Init) Run(out io.Writer) error {
 
 	adminKubeConfigPath := filepath.Join(kubeConfigDir, kubeadmconstants.AdminKubeConfigFileName)
 
+	// read the kubeadm log file and write a new one if it doesn't exist
```

Comment:
lets wrap this whole block like so:
```
if !i.dryRun {
	// read the kubeadm log file and write a new one if it doesn't exist
	data, err := kubeadmutil.ReadKubeadmLogFile()
	if err != nil {
		if err := kubeadmutil.WriteKubeadmLogFile("kubeadm init"); err != nil {
			glog.Fatalf("[init] cannot write kubeadm log file: %v", err)
		}
	}
	if data != "" {
		glog.Fatalf("[init] %q was already called on this machine", data)
	} else {
		glog.Fatal("[init] empty kubeadm log file")
	}
} else {
	fmt.Println("[init] would verify the kubeadm log file")
}
```

the idea here is that we don't want to handle the log file if dry running.
`join` doesn't have `--dry-run` so we only do this for `init`